File: frontend/src/model/ProgressDTO.ts
export interface ProgressDTO {
  id: string;
  attemptId: string;
  userFullName: string;
  courseId: string;
  courseName: string;
  courseCode: string;
  attemptCount: number;
  completedExercises: number;
  totalExercises: number;
  status: string;
  progressPercent: number;
  bestScore: number;
  lastActivityAt: string;
}

File: backend/src/main/java/com/example/iquiz/dto/ProgressDTO.java
package com.example.iquiz.dto;

import com.example.iquiz.enums.UserProgress;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class ProgressDTO {
    private UUID id;
    private UUID attemptId;
    private String userFullName;
    private UUID courseId;
    private String courseName;
    private String courseCode;
    private int attemptCount;
    private int completedExercises;
    private int totalExercises;
    private UserProgress status;
    private double progressPercent;
    private double bestScore;
    private LocalDateTime lastActivityAt;
}
